File: HL7 Programming using DotNet - Tutorial Series/CommonUtils/OurBase64Helper.cs
using System;
using System.IO;

namespace CommonUtils
{
    public class OurBase64Helper
    {
        public string ConvertToBase64String(FileInfo inputFile)
        {
            if (!inputFile.Exists)
                throw new FileNotFoundException($"The specified input file {inputFile.Name} does not exist");

            return Convert.ToBase64String(File.ReadAllBytes(inputFile.FullName));
        }

        public byte[] ConvertFromBase64String(string base64EncodedString)
        {
            if (string.IsNullOrEmpty(base64EncodedString))
                throw new ArgumentNullException(nameof(base64EncodedString), "You must supply data for Base64 decoding operation");

            if (base64EncodedString.Length % 4 != 0)
                throw new BadBase64EncodingException("The BASE-64 encoded data is not in correct form (divide by 4 resulted in a remainder)");

            try
            {
                return Convert.FromBase64String(base64EncodedString);
            }
            catch (Exception )
            {
                throw new ApplicationException("Unable to decode Base-64 string supplied for operation. Please check your inputs") ;
            }
        }
    }
}

File: HL7 Programming using Java - Tutorial Series/HL7 Programming using HAPI - Handling Binary Data/OurBase64Helper.java
package com.saravanansubramanian.hapihl7tutorial.handlingbinarydata;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.nio.file.Files;
import org.apache.commons.codec.binary.Base64;

public class OurBase64Helper {

	public String ConvertToBase64String(File inputFile) throws IOException
    {
        if (!inputFile.exists())
            throw new FileNotFoundException(String.format("The specified input file '%s' does not exist",inputFile.getAbsoluteFile()));

        return new String(Base64.encodeBase64(Files.readAllBytes(inputFile.toPath())));
    }

    public byte[] ConvertFromBase64String(String base64EncodedString) throws BadBase64EncodingException
    {
        if (base64EncodedString == null || base64EncodedString.length() == 0)
        	throw new IllegalArgumentException("You must supply byte string for Base64 decoding operation");
        
        if (base64EncodedString.length() % 4 != 0)
        	throw new BadBase64EncodingException("The BASE-64 encoded data is not in correct form (divide by 4 resulted in a remainder)");

        try {
			return Base64.decodeBase64(base64EncodedString);
		} catch (Exception e) {
			e.printStackTrace();
			throw new RuntimeException("Unable to decode Base-64 string supplied for operation. Please check your inputs");
		}
    }
}


